tests/test_branch_stats.py

```diff
@@ -72,45 +75,88 @@ def branch_length_diversity_window(ts, X, Y, windows):
     return out
 
 
-def branch_length_Y(ts, x, y, z):
+def branch_length_Y(ts, X, Y, Z, begin=0.0, end=None):
+    if end is None:
+        end = ts.sequence_length
     S = 0
     for tr in ts.trees():
-        xy_mrca = tr.mrca(x, y)
-        xz_mrca = tr.mrca(x, z)
-        yz_mrca = tr.mrca(y, z)
-        if xy_mrca == xz_mrca:
-            #   /\
-            #  / /\
-            # x y  z
-            S += path_length(tr, x, yz_mrca)*tr.length
-        elif xy_mrca == yz_mrca:
-            #   /\
-            #  / /\
-            # y x  z
-            S += path_length(tr, x, xz_mrca)*tr.length
-        elif xz_mrca == yz_mrca:
-            #   /\
-            #  / /\
-            # z x  y
-            S += path_length(tr, x, xy_mrca)*tr.length
-    return S/ts.sequence_length
-
-
-def branch_length_f4(ts, A, B, C, D):
+        if tr.interval[1] < begin:
+            continue
+        if tr.interval[0] > end:
```

Comment:
actually, both should be >= if both intervals (begin,end] and (interval[0], interval[1]] are half-open.  but, it won't make a difference since in the edge case `this_length` will be zero (changing anyhow)